forte/data/ontology/core.py

```diff
@@ -865,5 +867,167 @@ def index_key(self) -> int:
         return self.tid
 
 
+class Grid:
+    """
+    Regular grid with a grid configuration dependent on the image size.
+
+    We intialize and constrain the grid shape by the image shape during
+    initialization since usually there aren't many different image sizes in one dataset.
+
+    Args:
+        height: the number of grid cell per column, the unit is one grid cell.
+        width: the number of grid cell per row, the unit is one grid cell.
+        image_height: the number of pixels per column in the image.
+        image_width: the number of pixels per row in the image.
+    """
+
+    def __init__(
+        self,
+        height: int,
+        width: int,
+        image_height: int,
+        image_width: int,
+    ):
+        if height <= 0 or width <= 0:
+            raise ValueError(
+                f"height({height}) and "
+                f"width({width}) both must be larger than 0"
+            )
+        self._height = height
+        self._width = width
+        # based the image size and the grid shape
+        # we compute the height and width of grid cells
+        # for example, if the image size (image_height,image_width) is
+        #  (640, 480)
+        # and the grid shape (height, width) is (2, 3)
+        # the size of grid cells (self.c_h, self.c_w) will be (320, 240)
+        # if the resulting size of grid is not an integer, we round it up
+        # the last grid cell per row and column might be out of the image size
+        # since we constrain the maximum pixel locations by the image size
+        self.c_h, self.c_w = (
+            math.ceil(image_height / self._height),
+            math.ceil(image_width / self._width),
+        )
+        self.image_height = image_height
+        self.image_width = image_width
+        if image_height <= 0 or image_width <= 0:
+            raise ValueError(
+                "both image height and width must be positive"
+                f"but the image shape is {(image_height, image_width)}"
+                "please input a valid image shape"
+            )
+        if self.c_h <= 0 or self.c_w <= 0:
+            raise ValueError(
+                "cell height and width must be positive"
+                f"but the cell shape is {(self.c_h, self.c_w)}"
+                "please adjust image shape or grid shape accordingly"
+            )
+
+    def get_grid_cell(self, img_arr: np.ndarray, h_idx: int, w_idx: int):
+        """
+        Get the array data of a grid cell from image of the image payload index.
+        The array is a masked version of the original image, and it has
+        the same size of the image. The array entries that are not
+        within the grid cell will masked as zeros. The array entries that are
+        within the grid cell will be copied to the zeros numpy array.
```

Comment:
`copied to the zeros numpy array` is confusing, where did `the zeros numpy array` come from.